init/caddy-api.service

```diff
@@ -10,7 +10,8 @@
 [Unit]
 Description=Caddy
 Documentation=https://caddyserver.com/docs/
-After=network.target
+After=syslog.target network.target remote-fs.target nss-lookup.target network-online.target
```

Comment:
I don't think we should change more than is requisite -- wouldn't it be best to wait and see if there's actually a need for these other dependencies? Ideally we'd keep it as simple as possible.

Reply:
Those actually aren't dependencies, they just change the order if those services are registered. So it's safe to add in `After`. It makes sense though, because we need syslog to print our logs, nss for DNS when running, and possibly remote-fs if a custom storage directory is configured for running in a cluster.

Reply:
The freedesktop link provided [recommends just using network.target](https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/#conceptsinsystemd).

> Services using the network should hence simply place an After=network.target dependency in their unit files, and avoid any Wants=network.target or even Requires=network.target.

Furthermore, it cautions against using network-online.

>  It is strongly recommended not to pull in this target too liberally: for example network server software should generally not pull this in (since server software generally is happy to accept local connections even before any routable network interface is up), its primary purpose is network client software that cannot operate without network.

It goes into [further detail](https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/#whatdoesthismeanformeadeveloper) about how the best scenario is for programs "to be friendly to dynamically changing network configuration".  In other words, Caddy should start fine regardless of the network status, and gracefully accept the network connections coming online after the fact.  This allows systemd to minimize boot time by starting as many units in parallel as possible.

I believe the additional targets are not necessary.

- syslog.target: Not all systems have this (i.e. using journald), and the ones that do are already [required by systemd to support socket activation](https://www.freedesktop.org/wiki/Software/systemd/syslog/), so it's pointless to delay Caddy's startup until the target is fully available.
- remote-fs.target: Caddy shouldn't wait for network filesystems by default because that won't matter for most users.  See my point below about users overriding this if they know they need it.
- nss-lookup.target: This one makes more sense, but similar to network-online.target, Caddy should handle starting without this target and allow it to come online later.

I agree with Matt about keeping the default unit file minimal.  The more complicated you make your start up ordering and dependencies, the longer it will take to boot a system.  If a user does something custom like serving content from a network share or binding to an explicit IP address, it is trivial to extend the After directive with a local override.

```
# run this
systemctl edit caddy.service
# then add this text, which will be merged with the default unit file
[Unit]
After=remote-fs.target network-online.target
```

Adding a Requires is not advisable because in this instance it means that stopping the network would also stop Caddy, and starting Caddy with the network stopped would cause the network to be started.  This is not what most users would expect to happen if they are manually starting/stopping their network (for instance when troubleshooting), which is why other web servers don't do it either.